contracts/StablecoinConverter.sol

```diff
@@ -330,50 +330,50 @@ contract StablecoinConverter is EpochTokenLocker {
         uint128[] memory volumes,
         uint16[] memory tokenIdsForPrice
     ) internal {
-        previousSolution.batchId = batchIndex;
+        currentSolution.batchId = batchIndex;
         for (uint i = 0; i < owners.length; i++) {
-            previousSolution.trades.push(TradeData({
+            currentSolution.trades.push(TradeData({
                 owner: owners[i],
                 orderId: orderIds[i],
                 volume: volumes[i]
             }));
         }
-        previousSolution.tokenIdsForPrice = tokenIdsForPrice;
-        previousSolution.solutionSubmitter = msg.sender;
+        currentSolution.tokenIdsForPrice = tokenIdsForPrice;
+        currentSolution.solutionSubmitter = msg.sender;
     }
 
-    function undoPreviousSolution(uint32 batchIndex) internal {
-        if (previousSolution.batchId == batchIndex) {
-            for (uint i = 0; i < previousSolution.trades.length; i++) {
-                address owner = previousSolution.trades[i].owner;
-                uint orderId = previousSolution.trades[i].orderId;
+    function undoCurrentSolution(uint32 batchIndex) internal {
```

Comment:
Here, the naming undoCurrentSolution is much better, as we will not undo a solution from a previous batch